### PR TITLE
fix(Translations & Regression): changes for translation an regression…

### PR DESCRIPTION
**Pull request by EdmondCShaffer**
Related:  https://github.ibm.com/wiotp/Maximo-Asset-Monitor/issues/2521

**Summary**
Changes for translation bugs in monitor. Also one regression fix where callback was removed from bad merge in `CardEditForm.jsx`. Was purposely added [here](https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3454/files) and accidentally removed [here](https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3450/files)
- fix i18n strings in `DataSeriesFormContent.jsx`

**Change List (commits, features, bugs, etc)**

- add callback `onEditDataItem` back that was accidentally removed in a bad merge for `CardEditForm.jsx`. 
- add missing `translateWithId` callback to `ThresholdsFormItem.jsx`
- fix strings in `DataSeriesFormItemModal.jsx`

**Acceptance Test (how to verify the PR)**

- Go to [this](https://deploy-preview-3466--carbon-addons-iot-react.netlify.app/?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-dashboardeditor--with-initial-value) story.
- Click on "Default rendered card" and then go to side panel and choose 'v2 pressure' from the data item drop down
- click edit on v2 pressure that opens modal. 
- see translated string. 

**Regression Test (how to make sure this PR doesn't break old functionality)**

- all the dashboard editor feature should work with out any issue?

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Review by herleraja — APPROVED** [review]
LGTM

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx [hunk 1]
```diff
@@ -173,8 +173,10 @@ const defaultProps = {
     dataItemSource: 'Data item source',
     primaryButtonLabelText: 'Save',
     secondaryButtonLabelText: 'Cancel',
-    decimalPlaces: 'Decimal places',
+    decimalPlacesLabel: 'Decimal places',
```
Comment (davidicus) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal.jsx:176]: Should follow the convention of the others. 
```suggestion
    decimalPlacesLabelText: 'Decimal places',
```

**Review by davidicus — CHANGES_REQUESTED** [review]
small change request but otherwise looks good.

**Review by davidicus — APPROVED** [review]
I would argue that this would be an opportune time to change the wrong strings in`BarChartCardFormSettings` but I will approve to get this in.

**Pull request merged**